Challenge-4/Bot_2/src/helper.spec.ts

```diff
@@ -0,0 +1,133 @@
+  "function getUtilization() public view returns (uint)",
+  "function getSupplyRate(uint utilization) public view returns (uint64)",
+  "function getBorrowRate(uint utilization) public view returns (uint64)",
+];
+
+describe("Helper class test suite", () => {
+  let mockProvider: MockEthersProvider;
+  let Iface: ethers.utils.Interface = new ethers.utils.Interface(mockAbi);
+  let helper: Helper;
+  const usdcTokenAddress = "0xc3d688B66703497DAA19211EEdff47f25384cdc3";
+  let mockConfigProxy = "0x316f9708bb98af7da9c68c1c3b5e79039cd336e3";
+  beforeEach(() => {
+    mockProvider = new MockEthersProvider() as any;
+    helper = new Helper(mockProvider as any);
+  });
+
+  it("should return correct configuration data", async () => {
+    mockProvider.setNetwork(1);
+    mockProvider.addCallTo(mockConfigProxy, 0, Iface, "getConfiguration", {
+      inputs: [usdcTokenAddress],
+      outputs: [
+        {
+          governor: createAddress("0x123"),
+          pauseGuardian: createAddress("0x123"),
+          baseToken: createAddress("0x123"),
+          baseTokenPriceFeed: createAddress("0x123"),
+          extensionDelegate: createAddress("0x223"),
+          supplyKink: ethers.BigNumber.from(5000),
+          supplyPerYearInterestRateSlopeLow: ethers.BigNumber.from(60),
+          supplyPerYearInterestRateSlopeHigh: ethers.BigNumber.from(100),
+          supplyPerYearInterestRateBase: ethers.BigNumber.from(900),
+          borrowKink: ethers.BigNumber.from(4000),
+          borrowPerYearInterestRateSlopeLow: ethers.BigNumber.from(100),
+          borrowPerYearInterestRateSlopeHigh: ethers.BigNumber.from(200),
+          borrowPerYearInterestRateBase: ethers.BigNumber.from(300),
+          storeFrontPriceFactor: ethers.BigNumber.from(50),
+          trackingIndexScale: ethers.BigNumber.from(50),
+          baseTrackingSupplySpeed: ethers.BigNumber.from(50),
+          baseTrackingBorrowSpeed: ethers.BigNumber.from(50),
+          baseMinForRewards: ethers.BigNumber.from(50),
+          baseBorrowMin: ethers.BigNumber.from(50),
+          targetReserves: ethers.BigNumber.from(50),
+          assetConfigs: [
+            {
+              asset: usdcTokenAddress,
+              decimals: 18,
+              conversionFactor: ethers.BigNumber.from(1),
+            },
+          ],
+        },
+      ],
+    })
+    mockProvider.addCallTo(usdcTokenAddress, 0, Iface, "getUtilization", {
+      inputs: [],
+      outputs: [ethers.BigNumber.from(5000)],
+    })
+    mockProvider.addCallTo(usdcTokenAddress, 0, Iface, "getSupplyRate", {
+      inputs: [ethers.BigNumber.from(5000)],
+      outputs: [ethers.BigNumber.from(500000000)],
+    });
+    mockProvider.addCallTo(usdcTokenAddress, 0, Iface, "getBorrowRate", {
+      inputs: [ethers.BigNumber.from(5000)],
+      outputs: [ethers.BigNumber.from(500000000)],
```

Comment:
Unless the output of `getSupplyRate` and `getBorrowRate` are supposed to be the same, you should make them different, that way you're more certain that they're correct.

If they're returning the same value, the logic could return the other one and make it seem like the returned value is correct. E.g. you query `getBorrowRate` and compare it to `ethers.BigNumber.from(500000000)` and your `Helper` accidentally returns the value from `getSupplyRate`, but because they're the same, you wouldn't know your class is not working properly.

I hope I explained that well.